### PR TITLE
VACMS-6010 Refactor of clone code into drush job

### LINKED ISSUE #6010: Create new high capacity test setup
## Description

Setup a new test environment which has at least 48K nodes

## Acceptance Criteria
- [ ] `cms-test` staging will be setup with 48K nodes based upon the current data.

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [x] `Core Application Team`
- [ ] `Product Support Team`

### PR DESCRIPTION
**Pull request by indytechcook**
## Description

Closes #6010. 

Most of the code is refactoring the code from the `clone.php` file to a Drush job.  

Also, a more targeted clone is used. 

 > The main growth would be in the area of detail pages that need to be added for facilities and events and news, and staff.

per slack conversation: https://dsva.slack.com/archives/CT4GZBM8F/p1628022328100100

### CODE REVIEW

File: docroot/modules/custom/va_gov_clone/src/CloneManager.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  /**
+   * {@inheritDoc}
+   */
+  public function cloneEntity(EntityInterface $entity) : ?EntityInterface {
+    /** @var \Drupal\entity_clone\EntityClone\EntityCloneInterface $entity_clone_handler */
+    $entity_clone_handler = $this->entityTypeManager->getHandler($entity->getEntityTypeId(), 'entity_clone');
+
+    $duplicate = $entity->createDuplicate();
+    $this->eventDispatcher->dispatch(EntityCloneEvents::PRE_CLONE, new EntityCloneEvent($entity, $duplicate));
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_clone/src/CloneManager.php:66]: Are these arguments swapped?

Comment (ndouglas) [reply docroot/modules/custom/va_gov_clone/src/CloneManager.php:66]: https://symfony.com/blog/new-in-symfony-4-3-simpler-event-dispatching

Comment (indytechcook) [reply docroot/modules/custom/va_gov_clone/src/CloneManager.php:66]: I copy and pasted these from the `entity_clone` module :(

I didn't get any errors when running.  The Drupal version is different.

`Calling the Symfony\Component\EventDispatcher\EventDispatcherInterface::dispatch() method with a string event name as the first argument is deprecated in drupal:9.1.0, an Event object will be required instead in drupal:10.0.0. See https://www.drupal.org/node/3154407`

Comment (indytechcook) [reply docroot/modules/custom/va_gov_clone/src/CloneManager.php:66]: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21EventDispatcher%21ContainerAwareEventDispatcher.php/class/ContainerAwareEventDispatcher/9.1.x

Comment (ndouglas) [reply docroot/modules/custom/va_gov_clone/src/CloneManager.php:66]: you're telling PHP it's `\Symfony\Component\EventDispatcher\EventDispatcherInterface` tho which boils down to this: https://github.com/symfony/event-dispatcher-contracts/blob/main/EventDispatcherInterface.php

Comment (ndouglas) [reply docroot/modules/custom/va_gov_clone/src/CloneManager.php:66]: that's why PHPStan's pitching a fit

Comment (ndouglas) [reply docroot/modules/custom/va_gov_clone/src/CloneManager.php]: confused-math

https://www.drupal.org/project/drupal/issues/3055194

Comment (ndouglas) [reply docroot/modules/custom/va_gov_clone/src/CloneManager.php]: I guess do the thing that you started with and add it to the baseline (`lando phpstan --generate-baseline`)?  And I feel kinda like this existed in some indeterminate state where it wasn't going to cause an error or be any kind of problem until I got bored and wandered in to look at your PR and introduced my fractalicious chaos. ur welcome

File: docroot/modules/custom/va_gov_clone/drush.services.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+services:
+  va_gov_clone.commands:
+    class: Drupal\va_gov_clone\Commands\VaGovCloneCommands
+    arguments: ['@va_gov.clone.manager']
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_clone/drush.services.yml:4]: Why `va_gov_clone.commands` and `va_gov.clone.manager`?  I feel like it's more consistent to always do <module_name>.service.

File: docroot/modules/custom/va_gov_clone/src/CloneManager.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function cloneEntities(array $nodes) : void {
+    foreach ($nodes as $node) {
+      $this->cloneEntity($node);
+    }
+  }
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_clone/src/CloneManager.php:79]: $nodes or $entities?

File: docroot/modules/custom/va_gov_clone/src/CloneManagerInterface.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+   */
+  public function cloneAll(int $office_tid) : int;
+
+  /**
+   * Clone nodes.
+   *
+   * @param \Drupal\node\NodeInterface[] $nodes
+   *   Nodes to clone.
+   */
+  public function cloneEntities(array $nodes) : void;
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_clone/src/CloneManagerInterface.php:31]: Nodes or entities?

File: docroot/modules/custom/va_gov_clone/src/CloneManagerInterface.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  /**
+   * Clone All items.
+   *
+   * @param int $office_tid
+   *   The arguments passed from Drupal.
+   *
+   * @return int
+   *   The total count of content updated.
+   */
+  public function cloneAll(int $office_tid) : int;
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_clone/src/CloneManagerInterface.php:23]: I feel like the name of this method could be better.  Like `cloneOffice` or something?

File: docroot/modules/custom/va_gov_clone/va_gov_clone.info.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+name: Clone Content
+description: 'Clone content to generate fake content.  DO NOT ENABLE IN PRODUCTION'
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_clone/va_gov_clone.info.yml:2]: Could you do a thing in module_requirements to programatically prevent it from loading?

File: docroot/modules/custom/va_gov_clone/va_gov_clone.info.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+package: Custom
+type: module
+core_version_requirement: ^9
+hiden: true
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_clone/va_gov_clone.info.yml:6]: hiden?

File: docroot/modules/custom/va_gov_clone/src/CloneManagerInterface.php (hunk 5 above)
Comment (indytechcook) [reply docroot/modules/custom/va_gov_clone/src/CloneManagerInterface.php:23]: so it's actually a "section" vocabulary but naming things is hard

File: docroot/modules/custom/va_gov_clone/va_gov_clone.info.yml (hunk 6 above)
Comment (indytechcook) [reply docroot/modules/custom/va_gov_clone/va_gov_clone.info.yml:2]: `hidden` does that if it was spelled right https://www.drupal.org/docs/creating-custom-modules/let-drupal-know-about-your-module-with-an-infoyml-file

File: docroot/modules/custom/va_gov_clone/va_gov_clone.info.yml (hunk 7 above)
Comment (indytechcook) [reply docroot/modules/custom/va_gov_clone/va_gov_clone.info.yml:6]: https://www.drupal.org/docs/creating-custom-modules/let-drupal-know-about-your-module-with-an-infoyml-file

File: docroot/modules/custom/va_gov_clone/drush.services.yml (hunk 2 above)
Comment (indytechcook) [reply docroot/modules/custom/va_gov_clone/drush.services.yml:4]: Why look for reason in chaos?

File: phpstan-baseline.neon [hunk 8]
```diff
@@ -5,133 +5,13 @@ parameters:
 			count: 2
 			path: docroot/modules/custom/va_gov_backend/src/Form/ExclusionTypesAdminForm.php
 
-		-
```
Comment (indytechcook) [inline phpstan-baseline.neon:8]: I'm not sure what happened here.  I literally copied the file from `master` then added a single line.

Comment (ndouglas) [reply phpstan-baseline.neon]: ¯\\\_(ツ)\_/¯

**Comment by ndouglas** [comment]
Docs are clear.

Module didn't show up in module list, Drush command wasn't visible, etc 👍 

Command executed and claimed it did so successfully 😎 

Screen Shot 2021-08-10 at 10 32 21 AM

Screen Shot 2021-08-10 at 10 33 34 AM

**Review by ndouglas — APPROVED** [review]

**Pull request merged**